### PR TITLE
ElvaRivle/Misc/BugFixes

### PR DESCRIPTION
**Pull request by ElvaRivle**
ElvaRivle/Misc/BugFixes

--------------------------------------------------------------------
 MAIN
 * fixed all bugs that we discussed last time
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * ~~logout functionality~~

### CODE REVIEW

File: src/components/sidebar/Sidebar.js [hunk 1]
```diff
@@ -42,7 +44,16 @@ const Sidebar = ({actions}) => {
                     </p>
                 </div>
                 <div className="expand-user-info-section">
-                    <DownOutlined/>
+                    <Dropdown
+                        menu={{items: UserInfoOptions}}
+                        trigger={['click']}
+                    >
+                        <a onClick={(e) => e.preventDefault()}>
```
Comment (AntMarkoLovric) [inline src/components/sidebar/Sidebar.js]: move function outside of jsx

File: src/routing/protectedRoute.js [hunk 2]
```diff
@@ -13,7 +12,9 @@ const ProtectedRoute = () => {
     (
       <>
         <Sidebar />
-        <Outlet />
+        <div style={{width: "100%"}}>
```
Comment (AntMarkoLovric) [inline src/routing/protectedRoute.js]: please don't use inline style

File: src/routing/publicRoute.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const ProtectedRoute = () => {
+    //here is the core logic that prevents unauthorized access to routes
+    //each protected route should have this component as parent
+
+    let token = localStorage.getItem("accessToken")
+
+    return !token || token === "" ?
+        (
+            <>
+                <div style={{width: "100%"}}>
```
Comment (AntMarkoLovric) [inline src/routing/publicRoute.js]: same as the comment before regarding inline style

File: src/routing/publicRoute.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import {Navigate, Outlet, Route, Routes} from "react-router-dom";
+import Sidebar from "../components/sidebar/Sidebar";
+import React from "react";
+
+const ProtectedRoute = () => {
+    //here is the core logic that prevents unauthorized access to routes
+    //each protected route should have this component as parent
+
+    let token = localStorage.getItem("accessToken")
```
Comment (AntMarkoLovric) [inline src/routing/publicRoute.js:9]: what happens if you get invalid token

**Review by AntMarkoLovric — CHANGES_REQUESTED** [review]

File: src/components/sidebar/Sidebar.js (hunk 1 above)
Comment (becarselman) [inline src/components/sidebar/Sidebar.js]: you are resolve comment but don't fix anything

File: src/routing/protectedRoute.js (hunk 2 above)
Comment (becarselman) [inline src/routing/protectedRoute.js]: same

File: src/routing/publicRoute.js (hunk 3 above)
Comment (becarselman) [inline src/routing/publicRoute.js]: same

**Review by becarselman — CHANGES_REQUESTED** [review]

**Review by becarselman — APPROVED** [review]

**Pull request merged**